### PR TITLE
Add unlink button to escape out of adding link

### LINKED ISSUE #778: No way to abort making a link
If you select some text and click the link icon on the toolbar, there's no way to abort. If you click outside of the url input, the text is a link with no href, you have to click it again to get the control to remove the link.

I think it makes sense to have a cancel button on the url input, and it would be nice if pressing escape canceled too.

### PR DESCRIPTION
**Pull request by mkaz**
Add ability to cancel adding a link after you clicked the linked button. 
Fixes #778

### CODE REVIEW

**Comment by mkaz** [comment]
@jasmussen - first part is adding the extra unlink button when adding the link is clicked. Screenshot below, can I get your design eyes and let me know what you think. 

add-unlink-button

I'm trying now if I can grab the Esc key to cancel just adding the link and not kill the whole text block

**Comment by mkaz** [comment]
@notnownikki I think I have this working as you described the bug. A cancel button was added and the ability to escape when it opens without canceling the full block. Take a look and let me know what you think.

**Comment by jasmussen** [comment]
UX seems nice! 👍

**Review by notnownikki — DISMISSED** [review]
This doesn't work for me...

This is what I did:

I added a new text block and typed some text. I selected the text I wanted to turn into a link. I entered "http://google.com/" into the link box and pressed enter.

I expected the input to disappear at that point so I can carry on with the post, but it doesn't. Clicking back in the block doesn't make it go away either. So I pressed escape to get out of the link editing, and it removed the link completely.

I tried the same steps again, but tried clicking the "enter" icon instead, and that seemed to make the link but still left me in the input UI. I had to add another block, focus the new block, and go back to the original block to get out of it. When I did that, the text looked like a link, but clicking on it to bring back the link editing UI showed that there was no URL in the link.

File: blocks/editable/format-toolbar.js [hunk 1]
```diff
@@ -36,6 +37,11 @@ class FormatToolbar extends wp.element.Component {
 		this.dropLink = this.dropLink.bind( this );
 		this.submitLink = this.submitLink.bind( this );
 		this.updateLinkValue = this.updateLinkValue.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+	}
+
+	componentDidMount() {
+		document.addEventListener( 'keydown', this.onKeyDown );
```
Comment (aduth) [inline blocks/editable/format-toolbar.js]: If `keydown` the event we want, or do we want to wait for a complete press?

File: blocks/editable/format-toolbar.js [hunk 2]
```diff
@@ -44,6 +50,19 @@ class FormatToolbar extends wp.element.Component {
 		}
 	}
 
+	onKeyDown( keydown ) {
+		switch ( keydown.keyCode ) {
```
Comment (aduth) [inline blocks/editable/format-toolbar.js]: Maybe a simple `if` would be enough here for now.

File: blocks/editable/format-toolbar.js [hunk 3]
```diff
@@ -44,6 +50,19 @@ class FormatToolbar extends wp.element.Component {
 		}
 	}
 
+	onKeyDown( keydown ) {
```
Comment (aduth) [inline blocks/editable/format-toolbar.js]: By convention we might want to name this variable `event`.

**Comment by mkaz** [comment]
@notnownikki - there is an extra step after hitting (or clicking) enter that converts what was entered into a link and you can see it as a link (blue underline) in the input field. You then need to type Esc, or click out and it closes for me. So I'm not duplicating what it sounds like you described.

adding-link-example


The change I added was not to fix the above behavior, which seems to be working as designed. But to fix if you start a link and don't type anything and then hit escape to cancel making an empty link.

**Comment by notnownikki** [comment]
Ah, I see now, the problems I'm seeing are caused by selection code, totally unrelated to this branch. _Somehow_ I'm able to lock things up by the way I select the text to convert to a link, and end up with a URL input box that I can't get rid of.

But the ESC functionality works :)

**Pull request merged**